product_abc_classification/models/abc_classification_product_level.py

```diff
@@ -0,0 +1,178 @@
+    _rec_name = "level_id"
+
+    display_name = fields.Char(compute="_compute_display_name")
+
+    manual_level_id = fields.Many2one(
+        "abc.classification.level",
+        string="Manual classification level",
+        tracking=True,
+        domain="[('profile_id', '=', profile_id)]",
+    )
+    computed_level_id = fields.Many2one(
+        "abc.classification.level",
+        string="Computed classification level",
+        readonly=True,
+    )
+    level_id = fields.Many2one(
+        "abc.classification.level",
+        string="Classification level",
+        compute="_compute_level_id",
+        store=True,
+        domain="[('profile_id', '=', profile_id)]",
+    )
+    flag = fields.Boolean(
+        default=False,
+        compute="_compute_flag",
+        string="If True, this means that the manual classification is "
+        "different from the computed one",
+        store=True,
+        index=True,
+    )
+    product_id = fields.Many2one(
+        "product.product",
+        string="Product",
+        index=True,
+        required=True,
+        ondelete="cascade",
+    )
+    product_tmpl_id = fields.Many2one(
+        "product.template",
+        string="Product template",
+        index=True,
+        readonly=True,
+    )
+    # percentage
+    profile_id = fields.Many2one(
+        "abc.classification.profile",
+        string="Profile",
+        required=True,
+    )
+    profile_type = fields.Selection(
+        related="profile_id.profile_type",
+        readonly=True,
+        store=True,
+    )
+    allowed_profile_ids = fields.Many2many(
+        comodel_name="abc.classification.profile",
+        related="product_id.abc_classification_profile_ids",
+    )
+
+    _sql_constraints = [
+        (
+            "product_level_uniq",
+            "UNIQUE(profile_id, product_id)",
+            _("Only one level by profile by product allowed"),
+        )
+    ]
+
+    @api.constrains("computed_level_id", "manual_level_id", "product_id")
+    def _check_level(self):
+        for rec in self:
+            if not rec.computed_level_id and not rec.manual_level_id:
+                raise ValidationError(_("Classification level is mandatory"))
+            if (
+                rec.computed_level_id
+                and rec.computed_level_id.profile_id != rec.profile_id
+            ):
+                raise ValidationError(
+                    _(
+                        "Computed level must be in  the same classifiation "
+                        "profile as the one on the product level"
+                    )
+                )
+            if rec.manual_level_id and rec.manual_level_id.profile_id != rec.profile_id:
+                raise ValidationError(
+                    _(
+                        "Manual level must be in  the same classifiation "
+                        "profile as the one on the product level"
+                    )
+                )
+
+    @api.onchange("product_tmpl_id")
```

Comment:
It would be nice to convert this onchange to a computed field on the product_id.